src/test/scala/chiselTests/Module.scala

```diff
@@ -167,6 +168,71 @@ class ModuleSpec extends ChiselPropSpec with Utils {
           "a" -> m.a, "b" -> m.b))
     })
   }
+
+  property("DataMirror.modulePorts should replace deprecated <module>.getPorts") {
+    class MyModule extends Module {
+      val io = IO(new Bundle {
+        val in = Input(UInt(8.W))
+        val out = Output(Vec(2, UInt(8.W)))
+      })
+      val extra = IO(Input(UInt(8.W)))
+      val delay = RegNext(io.in)
+      io.out(0) := delay
+      io.out(1) := delay + extra
+    }
+    var mod: MyModule = null
+    ChiselStage.elaborate {
+      mod = new MyModule
+      mod
+    }
+    // Note that this is just top-level ports, Aggregates are not flattened
+    assert(DataMirror.modulePorts(mod) == Seq(
+      "clock" -> mod.clock,
+      "reset" -> mod.reset,
+      "io" -> mod.io,
+      "extra" -> mod.extra
+    ))
+    // Delete this when the deprecated API is deleted
+    // Note this also uses deprecated Port
+    import chisel3.internal.firrtl.Port
+    import SpecifiedDirection.{Input => IN, Unspecified}
+    assert(mod.getPorts == Seq(
+      Port(mod.clock, IN),
+      Port(mod.reset, IN),
+      Port(mod.io, Unspecified),
+      Port(mod.extra, IN)
+    )): @nowarn // delete when Port and getPorts become private
```

Comment:
Mega nit. Consider using the more human readable equivalent:

```scala
mod.getPorts should contain theSameElementsInOrderAs(/*...*/)
```

Ditto for the later test.